examples/release_highlights/plot_release_highlights_1_2_0.py

```diff
@@ -0,0 +1,79 @@
+# flake8: noqa
+"""
+=======================================
+Release Highlights for scikit-learn 1.2
+=======================================
+
+.. currentmodule:: sklearn
+
+We are pleased to announce the release of scikit-learn 1.2! Many bug fixes
+and improvements were added, as well as some new key features. We detail
+below a few of the major features of this release. **For an exhaustive list of
+all the changes**, please refer to the :ref:`release notes <changes_1_2>`.
+
+To install the latest version (with pip)::
+
+    pip install --upgrade scikit-learn
+
+or with conda::
+
+    conda install -c conda-forge scikit-learn
+
+"""
+
+# %%
+# Pandas output with `set_output` API
+# -----------------------------------
+# scikit-learn's transformers now support pandas output with the `set_output` API.
+# See :ref:`sphx_glr_auto_examples_miscellaneous_plot_set_output.py`, to learn more
+# about how to use the API.
+import numpy as np
+from sklearn.datasets import load_iris
+from sklearn.preprocessing import StandardScaler, KBinsDiscretizer
+from sklearn.compose import ColumnTransformer
+
+X, y = load_iris(as_frame=True, return_X_y=True)
+sepal_cols = ["sepal length (cm)", "sepal width (cm)"]
+petal_cols = ["petal length (cm)", "petal width (cm)"]
+
+preprocessor = ColumnTransformer(
+    [
+        ("scalar", StandardScaler(), sepal_cols),
+        ("kbin", KBinsDiscretizer(encode="ordinal"), petal_cols),
+    ],
+    verbose_feature_names_out=False,
+)
+preprocessor.set_output(transform="pandas")
+
+X_out = preprocessor.fit_transform(X)
+X_out.sample(n=5, random_state=0)
+
+# %%
+# Interaction constraints in Histogram-based Gradient Boosting Trees
+# ------------------------------------------------------------------
+# :class:`~ensemble.HistGradientBoostingRegressor` and
+# :class:`~ensemble.HistGradientBoostingClassifier` now supports interaction constraints
+# with the `interaction_cst` parameter. For details, see the
+# :ref:`User Guide <interaction_cst_hgbt>`. In the following example, features are not
+# allowed to interact.
+import matplotlib.pyplot as plt
+from sklearn.datasets import load_diabetes
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.inspection import PartialDependenceDisplay
+
+X, y = load_diabetes(return_X_y=True, as_frame=True)
+
+hist = HistGradientBoostingRegressor(interaction_cst=[[i] for i in range(X.shape[1])])
+hist.fit(X, y)
+
+_, ax = plt.subplots(figsize=(12, 4))
+PartialDependenceDisplay.from_estimator(hist, X, ["bp", "bmi"], n_cols=2, ax=ax)
+
```

Comment:
This is not easy to interpret this plot without additional explanations.

Wouldn't it make sense to also plot the PDB for the same model without the `interaction_cst`?

Reply:
I think adding an explanation would detract from the highlights. With that in mind, I removed the partial dependence curve. If a reader wants to learn more, there is a link to the User Guide.

REF: The [highlights for missing value support](https://scikit-learn.org/stable/auto_examples/release_highlights/plot_release_highlights_0_22_0.html#native-support-for-missing-values-for-gradient-boosting) in `HistGradientBoosting*` only shows that `fit` works.